tests/virt_autotest/virt_utils.pm

```diff
@@ -360,55 +394,58 @@ sub upload_supportconfig_log {
 # Download guest image and xml from a NFS location to local
 # the image and xml is coming from a guest installation testsuite
 # need set SKIP_GUEST_INSTALL=1 in the test suite settings
+# return the account of the guests downloaded
 # only available on x86_64
 sub download_guest_assets {
 
     # guest_pattern is a string, like sles-11-sp4-64, may or may not with pv or fv given.
-    my ($guest_pattern, $vm_xml_dir) = @_;
-
-    # list the guests matched the pattern
-    my $qa_guest_config_file = "/usr/share/qa/virtautolib/data/vm_guest_config_in_vh_update";
-    my $hypervisor_type      = get_var('SYSTEM_ROLE', '');
-    my $install_guest_list = script_output "source /usr/share/qa/virtautolib/lib/virtlib; get_vms_from_config_file $qa_guest_config_file $guest_pattern $hypervisor_type";
-    save_screenshot;
-    if ($install_guest_list eq '') {
-        record_soft_failure("Not found guest pattern $guest_pattern in $qa_guest_config_file");
-        return 1;
-    }
+    my ($expected_guests, $vm_xml_dir) = @_;
 
     # mount the remote NFS location of guest assets
     # OPENQA_URL="localhost" in local openQA instead of the IP, so the line below need to be turned on and set to the webUI IP when you are using local openQA
     # Tips: Using local openQA, you need "rcnfs-server start & vi /etc/exports; exportfs -r")
-    # set_var('OPENQA_URL', "your_ip");
+    # set OPENQA_URL="your_ip" on openQA web UI
     my $openqa_server = get_required_var('OPENQA_URL');
-    $openqa_server =~ s/^http:\/\///;
-    my $remote_export_dir = "/var/lib/openqa/factory/other/";
-    my $mount_point       = "/tmp/remote_guest";
+
+    # check if vm xml files have been uploaded
+    my @available_guests = ();
+    foreach my $guest (split "\n", $expected_guests) {
+        my $guest_asset = generate_guest_asset_name("$guest");
+        my $vm_disk_url = $openqa_server . "/assets/other/" . $guest_asset . '.disk';
+        $vm_disk_url =~ s#^(?!http://)(.*)$#http://$1#;    #add 'http://' at beginning if needed.
+        if (head($vm_disk_url)) {
```

Comment:
Which perl module has this "head" command ?

Reply:
use LWP::Simple 'head';